src/algebra/module.lean

```diff
@@ -320,3 +320,34 @@ theorem smul_mem_iff (r0 : r ≠ 0) : r • x ∈ p ↔ x ∈ p :=
  p.smul_mem r⟩
 
 end submodule
+
+namespace add_comm_monoid
+open add_monoid
+
+variables {M : Type*} [add_comm_monoid M]
+
+instance : semimodule ℕ M :=
+{ smul := smul,
+  smul_add := λ _ _ _, smul_add _ _ _,
+  add_smul := λ _ _ _, add_smul _ _ _,
+  mul_smul := λ _ _ _, mul_smul _ _ _,
+  one_smul := one_smul,
+  zero_smul := zero_smul,
+  smul_zero := smul_zero }
+
+end add_comm_monoid
+
+namespace add_comm_group
+
+variables {M : Type*} [add_comm_group M]
+
+instance : module ℤ M :=
+{ smul := gsmul,
+  smul_add := λ _ _ _, gsmul_add _ _ _,
+  add_smul := λ _ _ _, add_gsmul _ _ _,
+  mul_smul := λ _ _ _, gsmul_mul _ _ _,
```

Comment:
Note that `gsmul_mul` seems to be a confusing name. I would expect `mul_gsmul`. That would also be in line with the monoid case.